src/create-glamorous.js

```diff
@@ -114,6 +114,7 @@ function createGlamorous(splitProps) {
           {
             ...componentProperties,
             comp: newComp,
+            rootEl: newComp,
```

Comment:
This should probably be `rootEl: getRootEl(newComp)` just in case someone passes `withComponent(SomeGlamorousComponent)` :+1:

Reply:
done

Reply:
hmm... `getRootEl` might do more harm than help there. Could you try if this example works for your code? (bigger font size should be applied) https://codesandbox.io/s/pkr1rn24vx

Reply:
Hmm... I'm not sure I understand. I think your codesandbox simply reveals the bug that would be fixed by these changes.

Reply:
@kentcdodds I didn't have time to explain this properly yesterday, so here it goes:

Normally, when you pass a glamorous component to a `glamourous()`, it 'merges' all the glamorous props so that there is only one instance of a glamorous component. `withComponent` works differently - when you pass a glamorous component to it there will be two instances - one glamorous instance rendering the other.

`rootEl` purpose is to decide what props should be filtered out (props that are invalid for a given `rootEl` won't be passed to the rendered component). When you assign `rootEl` to a wrapped non-glamorous component (mosty html element) all custom props will be filtered out.

This causes problems when this inner glamorous component (the one given to `withComponent`) relies on custom props for dynamic styles (it won't be able to access the custom props).

In summary: when passing glamorous component to `withComponent`, the component that will be directly rendered is the glamorous component itself so it should be assigned to `rootEl`.

tldr;
this works in 4.11.2 https://codesandbox.io/s/pkr1rn24vx
but doesn't in the latest (4.11.4)  https://codesandbox.io/s/n0yrxyrnl

Reply:
I think I understand. Could you perhaps open a pull request which adds a test for your scenario and see if we can satisfy both cases? If not then we can have a discussion about which to favor...